src/lseq/mod.rs

```diff
@@ -55,48 +55,48 @@ pub enum Op {
     },
 }
 
-impl LSeq {
+impl<T : Clone> LSeq<T> {
     /// Create an LSeq for the empty string
     pub fn new(id: u32) -> Self {
-        LSeq { text: Vec::new(), gen: IdentGen::new(id), clock: 0 }
+        LSeq { seq: Vec::new(), gen: IdentGen::new(id), clock: 0 }
     }
-    fn do_insert(&mut self, ix: Identifier, clock: u64, site_id: u32, c: char) {
+    fn do_insert(&mut self, ix: Identifier, clock: u64, site_id: u32, c: T) {
         // Inserts only have an impact if the identifier is in the tree
-        if let Err(res) = self.text.binary_search_by(|e| e.0.cmp(&ix)) {
-            self.text.insert(res, (ix, clock, site_id, c));
+        if let Err(res) = self.seq.binary_search_by(|e| e.0.cmp(&ix)) {
+            self.seq.insert(res, (ix, clock, site_id, c));
         }
     }
 
     fn do_delete(&mut self, ix: &Identifier) {
         // Deletes only have an effect if the identifier is already in the tree
-        if let Ok(i) = self.text.binary_search_by(|e| e.0.cmp(&ix)) {
-            self.text.remove(i);
+        if let Ok(i) = self.seq.binary_search_by(|e| e.0.cmp(&ix)) {
+            self.seq.remove(i);
         }
     }
 
     /// Apply an operation to an LSeq instance.
-    pub fn apply(&mut self, op: &Op){
+    pub fn apply(&mut self, op: &Op<T>){
```

Comment:
I think consuming the `Op` would make more sense. The caller should `clone` if they need a copy of it